### ISSUE TITLE
[BUG] RHEL-related automation should not inject log-collector hostname into /etc/rsyslog.conf

### ISSUE THREAD

**Issue by ferricoxide** [ComplianceAsCode/content#11812]
#### Share the context

We use this project's content across several customers' projects. Initially discovered the problem last year, but reporting the issue "fell off the stack". Recently, began receiving complaints from another customer that they were seeing "host not found" entries in their system logs for the `logcollector` host specified in the post-hardening `/etc/rsyslog.conf` configuration.

#### Description of problem:

Looks like this project's remediation-automation is injecting `@@logcollector` into the hardened-systems' /etc/rsyslog.conf` configuration. At bare minimum, this results in nuisance "host not found" error-types. Also poses a potential security-risk: if someone injects `logcollector` into the DNS domain the hardened system is configured into, log-offload attempts could be made to a bogus/unsanctioned rsyslog. Security-concern aside, if someone  injects `logcollector` into the DNS domain the hardened system is configured into, but doesn't stand up a collector, the "host not found" errors will be replaced with "cannot connect" error-types

#### Proposed change:

Centralize log-collection is inherently site-specific. In many cases, other services (e.g., Splunk, CloudWatch Agent, etc.) are used for offload rather than syslog. As such, the automation should include no remediative-actions that include injecting a logging-destination into the hardened-hosts' `/etc/rsyslog.conf` configurations.

CC: @lorengordon and @eemperor

**Comment by marcusburghardt** [comment]
It is likely related to the rule `rsyslog_remote_loghost`.

**Comment by ferricoxide** [comment]
@marcusburghardt sed:
> It is likely related to the rule `rsyslog_remote_loghost`.

Yeah, when I'd generated the BASH fix-script from the remediation-content, I believe that was the title of the block in the generated-script. We don't generally use tayloring-files to prevent scans/remediations from running.

That usage-method aside, I still feel that, given the inherently-local nature of remote log-offload, having the default content inject a (generally spurious) remote-logging entry into hardened-hosts' `rsyslog.conf` files doesn't make a ton of sense.

**Comment by marcusburghardt** [comment]
This issue is likely related: https://github.com/ComplianceAsCode/content/issues/8018
@ferricoxide , would you like to propose a PR improving this rule?

**Comment by ferricoxide** [comment]
@marcusburghardt 
I've mostly used this project's content in a fairly static way: is there a way to add usage of externalities to guide the setup that the `… --remediate …` automation does? Otherwise, the only PR I'd submit would be one that simply avoids tacking on the offending line.

**Comment by mpurg** [comment]
I agree with @ferricoxide on this. The remediation should not inject an invalid hostname by default or inject a configuration which is overly opinionated and most likely not how the end-user wants to setup their log offloading (maybe the user wants omrelp with tls and queues instead).

The below CIS recommendations have a Manual assessment status, so at least for these, there should be no need to do an automated assessment or remediation:
- RHEL8 v3.0.0: 5.1.1.6 Ensure rsyslog is configured to send logs to a remote log host
- RHEL9 v2.0.0: 6.2.3.6 Ensure rsyslog is configured to send logs to a remote log host
- Ubuntu 22.04 v1.0.0: 4.2.2.6 Ensure rsyslog is configured to send logs to a remote log host
- Ubuntu 24.04 v1.0.0: 6.1.3.6 Ensure rsyslog is configured to send logs to a remote log host
- SLE15 v2.0.0: 6.2.3.1.1 Ensure rsyslog is configured to send logs to a remote log host
- OL9 v2.0.0: 6.2.3.6 Ensure rsyslog is configured to send logs to a remote log host

I would propose to disable the remediation on the above systems or, if backwards compatibility is desired, the remediation could be modified to do nothing when the value is set to the default  `logcollector`.
